### PR TITLE
Update this doc to reflect a recent change on ValidationContext.MemberName Property

### PR DESCRIPTION
**Pull request by HongGit**
## Summary

In October .NET framework update there was an update to the ValidationContext.MemberName Property.  Update this doc to reflect the change.

### CODE REVIEW

File: includes/migration-guide/runtime/asp/aspnet-validationcontextmembername-not-null-when-using-custom.md [hunk 1]
```diff
@@ -2,8 +2,8 @@
...
+|Details|In .NET Framework 4.7.2 and earlier versions, when using a custom <xref:System.ComponentModel.DataAnnotations.ValidationAttribute?displayProperty=nameWithType>, the <xref:System.ComponentModel.DataAnnotations.ValidationContext.MemberName?displayProperty=nameWithType> property returns <code>null</code>.  Starting with [.NET Framework October 2019 Preview of Quality Rollup](https://devblogs.microsoft.com/dotnet/net-framework-october-2019-preview-of-quality-rollup/) for .NET Framework 4.8, it could return the member name after adding the following settings in the web config file:|
```
Comment (Rick-Anderson) [inline includes/migration-guide/runtime/asp/aspnet-validationcontextmembername-not-null-when-using-custom.md]: |Details|In .NET Framework 4.7.2 and earlier versions, when using a custom <xref:System.ComponentModel.DataAnnotations.ValidationAttribute?displayProperty=nameWithType>, the <xref:System.ComponentModel.DataAnnotations.ValidationContext.MemberName?displayProperty=nameWithType> property returns <code>null</code>.  Starting with [.NET Framework October 2019 Preview of Quality Rollup](https://devblogs.microsoft.com/dotnet/net-framework-october-2019-preview-of-quality-rollup/) for .NET Framework 4.8, it returns the member name when using the following settings in the *web.config* file:|

File: includes/migration-guide/runtime/asp/aspnet-validationcontextmembername-not-null-when-using-custom.md [hunk 2]
```diff
@@ -2,8 +2,8 @@
...
+|Suggestion|<pre><code class="lang-xml">&lt;configuration&gt;&#13;&#10;&lt;appSettings&gt;&#13;&#10;...&#13;&#10;&lt;add key=&quot;aspnet:GetValidationMemberName&quot;  value=&quot;true&quot;/&gt;&#13;&#10;...&#13;&#10;&lt;/appSettings&gt;&#13;&#10;&lt;/configuration&gt;&#13;&#10;</code></pre>In .NET Framework 4.8 version prior to the October update, this property returns a non-null value for a custom `ValidationAttribute` by default.
```
Comment (Rick-Anderson) [inline includes/migration-guide/runtime/asp/aspnet-validationcontextmembername-not-null-when-using-custom.md]: |Suggestion|<pre><code class="lang-xml">&lt;configuration&gt;&#13;&#10;&lt;appSettings&gt;&#13;&#10;...&#13;&#10;&lt;add key=&quot;aspnet:GetValidationMemberName&quot;  value=&quot;true&quot;/&gt;&#13;&#10;...&#13;&#10;&lt;/appSettings&gt;&#13;&#10;&lt;/configuration&gt;&#13;&#10;</code></pre>In .NET Framework 4.8 versions prior to the October update, this property returns a non-null value for a custom `ValidationAttribute` by default.

**Review by Rick-Anderson — APPROVED** [review]

File: includes/migration-guide/runtime/asp/aspnet-validationcontextmembername-not-null-when-using-custom.md (hunk 1 above)
Comment (mairaw) [reply includes/migration-guide/runtime/asp/aspnet-validationcontextmembername-not-null-when-using-custom.md]: should this be inside a suggestion Rick?

Comment (Rick-Anderson) [reply includes/migration-guide/runtime/asp/aspnet-validationcontextmembername-not-null-when-using-custom.md]: I missed the suggestion. I've never seen suggestion. I don't think so. What should be there?

https://review.docs.microsoft.com/en-us/dotnet/framework/migration-guide/runtime/4.0-4.8?branch=pr-en-us-15814#aspnet-validationcontextmembername-is-not-null-when-using-custom-dataannotationsvalidationattribute

Comment (HongGit) [reply includes/migration-guide/runtime/asp/aspnet-validationcontextmembername-not-null-when-using-custom.md]: updated

File: includes/migration-guide/runtime/asp/aspnet-validationcontextmembername-not-null-when-using-custom.md (hunk 2 above)
Comment (HongGit) [reply includes/migration-guide/runtime/asp/aspnet-validationcontextmembername-not-null-when-using-custom.md]: done.

File: includes/migration-guide/runtime/asp/aspnet-validationcontextmembername-not-null-when-using-custom.md (hunk 1 above)
Comment (mairaw) [reply includes/migration-guide/runtime/asp/aspnet-validationcontextmembername-not-null-when-using-custom.md]: This is the suggestion button just for reference. Then the person can just commit your suggestion.

image

Comment (HongGit) [reply includes/migration-guide/runtime/asp/aspnet-validationcontextmembername-not-null-when-using-custom.md]: I see, I didn't see the commit button for some reason so I submitted the change.

**Comment by mairaw** [comment]
The suggestion section now looks a bit odd since there's no intro in the suggestion cell.
Compare
https://review.docs.microsoft.com/en-us/dotnet/framework/migration-guide/runtime/4.0-4.8?branch=pr-en-us-15814#aspnet-validationcontextmembername-is-not-null-when-using-custom-dataannotationsvalidationattribute

with 
https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/runtime/4.0-4.8#aspnet-validationcontextmembername-is-not-null-when-using-custom-dataannotationsvalidationattribute

**Comment by mairaw** [comment]
The title of the breaking change is 
"ASP.NET ValidationContext.MemberName is not NULL when using custom DataAnnotations.ValidationAttribute"

So with the change made on the last quality rollup, it seems is null by default unless you do the config change. But for RTM, you need to do the config change to get a null value?

**Comment by mairaw** [comment]
@HongGit question about this PR. Since the default behavior now is opt-in, is this still a breaking change? The title of the breaking change is "ASP.NET ValidationContext.MemberName is not NULL when using custom DataAnnotations.ValidationAttribute", but it is now back to being null by default.

/cc @PriyaPurkayastha

**Comment by PriyaPurkayastha** [comment]
As I understand from reading this through, for .NET Framework 4.7.2 and earlier, the value returned is null and .NET 4.8 used to return non-null. With the .NET 4.8 Oct update, for .NET 4.8, null will be returned by default unless the customer chooses to opt-in in which case they will get a non-null value. Even though the default behavior of returning null for .NET 4.8 is the now the same as .NET 4.7.2 and earlier, we will need to document this as a breaking change since we have to document the compat opt-in switch which can determine what should be returned for .NET 4.8.

**Comment by mairaw** [comment]
Sounds good @PriyaPurkayastha. I'll make suggestions to make improvements to the text similar to what we've done to the API docs.

File: includes/migration-guide/runtime/asp/aspnet-validationcontextmembername-not-null-when-using-custom.md [hunk 3]
```diff
@@ -2,8 +2,8 @@
...
+|Details|In .NET Framework 4.7.2 and earlier versions, when using a custom <xref:System.ComponentModel.DataAnnotations.ValidationAttribute?displayProperty=nameWithType>, the <xref:System.ComponentModel.DataAnnotations.ValidationContext.MemberName?displayProperty=nameWithType> property returns <code>null</code>.  Starting with [.NET Framework October 2019 Preview of Quality Rollup](https://devblogs.microsoft.com/dotnet/net-framework-october-2019-preview-of-quality-rollup/) for .NET Framework 4.8, it returns the member name when using the following settings in the *web.config* file:|
```
Comment (mairaw) [inline includes/migration-guide/runtime/asp/aspnet-validationcontextmembername-not-null-when-using-custom.md]:
```suggestion
|Details|In .NET Framework 4.7.2 and earlier versions, when using a custom <xref:System.ComponentModel.DataAnnotations.ValidationAttribute?displayProperty=nameWithType>, the <xref:System.ComponentModel.DataAnnotations.ValidationContext.MemberName?displayProperty=nameWithType> property returns `null`. In .NET Framework 4.8 version prior to the October 2019 update, it returns the member name. Starting with [.NET Framework October 2019 Preview of Quality Rollup](https://devblogs.microsoft.com/dotnet/net-framework-october-2019-preview-of-quality-rollup/) for .NET Framework 4.8, it returns `null` by default, but you can opt in to return the member name instead. |
```

File: includes/migration-guide/runtime/asp/aspnet-validationcontextmembername-not-null-when-using-custom.md [hunk 4]
```diff
@@ -2,8 +2,8 @@
...
+|Suggestion|<pre><code class="lang-xml">&lt;configuration&gt;&#13;&#10;&lt;appSettings&gt;&#13;&#10;...&#13;&#10;&lt;add key=&quot;aspnet:GetValidationMemberName&quot;  value=&quot;true&quot;/&gt;&#13;&#10;...&#13;&#10;&lt;/appSettings&gt;&#13;&#10;&lt;/configuration&gt;&#13;&#10;</code></pre>In .NET Framework 4.8 versions prior to the October update, this property returns a non-null value for a custom `ValidationAttribute` by default.
```
Comment (mairaw) [inline includes/migration-guide/runtime/asp/aspnet-validationcontextmembername-not-null-when-using-custom.md]:
```suggestion
|Suggestion|Add the following setting to your *web.config* file for teh property to return the member name [.NET Framework October 2019 Preview of Quality Rollup](https://devblogs.microsoft.com/dotnet/net-framework-october-2019-preview-of-quality-rollup/) for .NET Framework 4.8 and later versions:<pre><code class="lang-xml">&lt;configuration&gt;&#13;&#10;&lt;appSettings&gt;&#13;&#10;...&#13;&#10;&lt;add key=&quot;aspnet:GetValidationMemberName&quot;  value=&quot;true&quot;/&gt;&#13;&#10;...&#13;&#10;&lt;/appSettings&gt;&#13;&#10;&lt;/configuration&gt;&#13;&#10;</code></pre>In .NET Framework 4.8 version prior to the October 2019 update,  adding this to your *web.config* file restores the previous behavior and the property returns `null`.|
```

**Review by mairaw — CHANGES_REQUESTED** [review]
@HongGit let me know what you think of my proposal. I think this would address some inconsistencies I'm seeing on your PR right now.

**Review by mairaw — APPROVED** [review]

**Pull request merged**